### ISSUE TITLE
[CI] S3CleanupTests testCleanup failing

### ISSUE THREAD

**Issue by jdconrad** [elastic/elasticsearch#70941]
**Build scan:**
https://gradle-enterprise.elastic.co/s/czrlkop4uz5ry/tests/:x-pack:snapshot-tool:qa:s3:s3ThirdPartyTest/org.elasticsearch.snapshots.S3CleanupTests/testCleanup

**Reproduction line:**
`./gradlew ':x-pack:snapshot-tool:qa:s3:s3ThirdPartyTest' --tests "org.elasticsearch.snapshots.S3CleanupTests.testCleanup" -Dtests.seed=39414F5E730936BB -Dtests.security.manager=false -Dtests.locale=ko-KR -Dtests.timezone=Europe/Vilnius -Druntime.java=8`

**Applicable branches:**
7.12

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.snapshots.S3CleanupTests&tests.test=testCleanup

**Failure excerpt:**
```
  org.elasticsearch.repositories.RepositoryException: [test-repo] Could not read repository data because the contents of the repository do not match its expected state. This is likely the result of either concurrently modifying the contents of the repository by a process other than this cluster or an issue with the repository's underlying storage. The repository has been disabled to prevent corrupting its contents. To re-enable it and continue using it please remove the repository from the cluster and add it again to make the cluster recover the known state of the repository from its physical contents. The last cluster to write to this repository was [_na_] at generation [-1].

    at __randomizedtesting.SeedInfo.seed([39414F5E730936BB:696B1CF00AA4A60E]:0)
    at org.elasticsearch.repositories.blobstore.BlobStoreRepository.corruptedStateException(BlobStoreRepository.java:1645)
    at org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$doGetRepositoryData$52(BlobStoreRepository.java:1572)
    at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:117)
    at org.elasticsearch.repositories.blobstore.BlobStoreRepository$5.clusterStateProcessed(BlobStoreRepository.java:1690)
    at org.elasticsearch.cluster.service.MasterService$SafeClusterStateTaskListener.clusterStateProcessed(MasterService.java:523)
    at org.elasticsearch.cluster.service.MasterService$TaskOutputs.lambda$processedDifferentClusterState$1(MasterService.java:410)
    at java.util.ArrayList.forEach(ArrayList.java:1259)
    at org.elasticsearch.cluster.service.MasterService$TaskOutputs.processedDifferentClusterState(MasterService.java:410)
    at org.elasticsearch.cluster.service.MasterService.onPublicationSuccess(MasterService.java:270)
    at org.elasticsearch.cluster.service.MasterService.publish(MasterService.java:262)
    at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:239)
    at org.elasticsearch.cluster.service.MasterService.access$000(MasterService.java:62)
    at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:140)
    at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139)
    at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:241)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:204)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)

    Caused by: org.elasticsearch.repositories.RepositoryException: [test-repo] could not read repository data from index blob

      at org.elasticsearch.repositories.blobstore.BlobStoreRepository.getRepositoryData(BlobStoreRepository.java:1717)
      at org.elasticsearch.repositories.blobstore.BlobStoreRepository.doGetRepositoryData(BlobStoreRepository.java:1528)
      at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:732)
      at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
      at java.lang.Thread.run(Thread.java:748)

      Caused by: java.nio.file.NoSuchFileException: Blob object [7.12/index-7] not found: The specified key does not exist. (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey; Request ID: X4W7WCB59NDF4D24; S3 Extended Request ID: YrNOlhHBk0N0jAbY/eMqu4p7J7J5+hUxVwz3Xdali+SORhoIZ8Sxah4M7F2/TnplupdiHecvm1g=)

        at org.elasticsearch.repositories.s3.S3RetryingInputStream.openStream(S3RetryingInputStream.java:91)
        at org.elasticsearch.repositories.s3.S3RetryingInputStream.<init>(S3RetryingInputStream.java:73)
        at org.elasticsearch.repositories.s3.S3RetryingInputStream.<init>(S3RetryingInputStream.java:56)
        at org.elasticsearch.repositories.s3.S3BlobContainer.readBlob(S3BlobContainer.java:90)
        at org.elasticsearch.repositories.blobstore.BlobStoreRepository.getRepositoryData(BlobStoreRepository.java:1703)
        at org.elasticsearch.repositories.blobstore.BlobStoreRepository.doGetRepositoryData(BlobStoreRepository.java:1528)
        at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:62)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:732)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

```

**Comment by original-brownbear** [comment]
This is caused by the repo-test-kit S3 tests running parallel to these tests, both of them using the same bucket. I'll fix it the same way we fixed a similar issue for the repo cleanup tooling tests.

![image](https://user-images.githubusercontent.com/6490959/113276288-fc1c2f80-92df-11eb-91ef-46b308a081fc.png)

**Comment by original-brownbear** [comment]
closed via #71173
